main.go

```diff
@@ -31,44 +28,52 @@ type Exporter struct {
 	mutex   sync.RWMutex
 }
 
-// Get Project-specific quotas
-func (e *Exporter) getProjectQuotas(ch chan<- prometheus.Metric) {
+// scrape connects to the Google API to retreive quota statistics and record them as metrics.
+func (e *Exporter) scrape() (up float64, prj *compute.Project, rgl *compute.RegionList) {
+
 	project, err := e.service.Projects.Get(e.project).Do()
 	if err != nil {
-		log.Fatalf("Unable to query API: %v", err)
+		log.Errorf("Failure when querying project quotas: %v", err)
+		return 0, nil, nil
+	}
+
+	regionList, err := e.service.Regions.List(e.project).Do()
+	if err != nil {
+		log.Errorf("Failure when querying region quotas: %v", err)
+		return 0, nil, nil
 	}
+
+	return 1, project, regionList
+}
+
+// Describe is implemented with DescribeByCollect. That's possible because the
+// Collect method will always return the same metrics with the same descriptors.
+func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(e, ch)
+}
+
+// Collect will run each time the exporter is polled and will in turn call the
+// Google API for the required statistics.
+func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
+	e.mutex.Lock() // To protect metrics from concurrent collects.
+	defer e.mutex.Unlock()
+
+	up, project, regionList := e.scrape()
```

Comment:
to be more "GOish" should you return `err` instead of `up` and have and use that err to define the up metric for this scrape ?